tests/test_utils/soc_desc_test_utils.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include <string>
+
+
+namespace test_utils {
+
+static std::size_t get_num_harvested(std::size_t harvesting_mask) {
+    std::size_t num_harvested = 0;
+    while (harvesting_mask > 0) {
+        if (harvesting_mask & 1) {
+            num_harvested++;
+        }
+        harvesting_mask >>= 1;
+    }
+    return num_harvested;
```

Comment:
```suggestion
    const std::bit_set<sizeof(harvesting_mask) * CHAR_BIT> harvesting_bits{harvesting_mask};
    return harvesting_bits.count();
```

Reply:
@joelsmithTT any reasoning behind thumbs down for this? @patrickroberts is bit_set more performant here?

Reply:
It's less readable.  The loop is obvious.  If brevity is important, you can say `return __builtin_popcountll(bits); // counts the 1 bits`

Reply:
Went with `__builtin_popcount`